### PR TITLE
feature/multi-element-action

### LINKED ISSUE #456: Poder deletar varios componentes apertando o botão [delete]
### Sua sugestão de melhoria é relacionada a algum problema?

Sim!! o problema é que desperdiço tempo clicando e deletando componente por componente.

---

### Descreva sua ideia de como resolver o problema

Poder deletar varios componentes apertando o botão [delete]
Gostei muito dos outros atalhos de Ctrl+S e Esc, agora só falta poder deletar com o delete.

---

### LINKED ISSUE #473: Deletar
Não existe a possibilidade de deletar quando se está com mais de um item selecionado. Somente um por um

### PR DESCRIPTION
**Pull request by miltonbsn**
# Summary

- Delete all selected elements #456 
- Copy and paste all selected elements [attention]

## Copy and paste allowed only on conceptual module.

Logic relations have deep internal connections (AKA foreign key), to copy and paste elements we have to update all these references and although not impossible, it is very complex and could lead to bugs.

In order to inform users about this limitation, the following message is now visible when users try to copy + paste elements in a logic model

image

## Unrelated fixes

- Fix email type on signup page
--- 

Close #456 
Close #473

### CODE REVIEW

**Comment by idmarjr** [comment]
> _Éx um monxtro!_

I did some tests here and it's working smoothly.  
Do you want to go above and beyond?

1. Once only one element is selected, show the selection mark too (The one with green border)
2. Allow user to select multiple elements one-by-one when holding `cmd` / `ctrl` and clicking in each element.

**Comment by miltonbsn** [comment]
> > _Éx um monxtro!_
> 
> I did some tests here and it's working smoothly. Do you want to go above and beyond?
> 
> 1. Once only one element is selected, show the selection mark too (The one with green border)
> 2. Allow user to select multiple elements one-by-one when holding `cmd` / `ctrl` and clicking in each element.

1) could not do it because the actions conflict
2) I tried but did not work after spending good amount of time. Will try more but wont hold the PR

**Comment by idmarjr** [comment]
@miltonbsn  regarding the limitation to copy + paste in Logic models, I have the following question:

Considering Foreign key as the limitation factor, can we enable c+p if a table still don't have stablished relationships yet?  
Meaning, should we enable user co c+p tables that dont have a connection with any other table?

File: app/i18n/languages/en.js [hunk 1]
```diff
@@ -157,5 +157,6 @@ export default {
 	'Delete account model error': 'It was not possible to delete your account because you have models. Please remove all your models before deleting your account.',
 	'Forbidden access': 'Forbidden access',
 	'Looks like this model does not exist or you don\'t have access to it.': 'Looks like this model does not exist or you don\'t have access to it.',
-	'Back to models list': 'Back to models list'
-};
\ No newline at end of file
+	'Back to models list': 'Back to models list',
+	'Copy and paste is not allowed on this module due to complexity when updating references': 'Copy and paste is not allowed on this module due to complexity when updating references'
```
Comment (idmarjr) [inline app/i18n/languages/en.js]: Is this copy being used?

**Comment by idmarjr** [comment]
On more thing...  
Is this branch mixing two different fixes?

Branch name is `feature/validate-mail` and I also see some changes related to that... 😓

**Comment by miltonbsn** [comment]
> @miltonbsn regarding the limitation to copy + paste in Logic models, I have the following question:
> 
> Considering Foreign key as the limitation factor, can we enable c+p if a table still don't have stablished relationships yet? Meaning, should we enable user co c+p tables that dont have a connection with any other table?

Yesss, I was a bit lazy but will do it :sweat_smile:

**Comment by miltonbsn** [comment]
> On more thing... Is this branch mixing two different fixes?
> 
> Branch name is `feature/validate-mail` and I also see some changes related to that... 😓

Yes, I started doing one thing and had this idea and delivered two. Sorry and forgive me this time :pray:

File: app/i18n/languages/en.js (hunk 1 above)
Comment (miltonbsn) [reply app/i18n/languages/en.js]: Good catch! It was my first version but found too long, changed and forgot this here.

The new one will be:

EN: `Copy is not allowed on this module when element has references.`
PT-BR: `Copiar não é permitido nesse módulo quando elementos já estiverem conectados.`

What do you think?

**Comment by idmarjr** [comment]
> Yes, I started doing one thing and had this idea and delivered two. Sorry and forgive me this time 🙏

judjing-you

File: app/i18n/languages/en.js (hunk 1 above)
Comment (idmarjr) [reply app/i18n/languages/en.js]: Looks good to me. Will you update it here or in a new branch?

Comment (miltonbsn) [reply app/i18n/languages/en.js]: Here, I am working on it right now. Let's try merge this today.

**Comment by miltonbsn** [comment]
@idmarjr we are good to go again. I did a mistake when updating this branch but i will fix when merging this PR.

**Comment by miltonbsn** [comment]
@idmarjr wait, it is worst than I expected, i will open another PR

**Pull request closed without merging**